js/Common.js

```diff
@@ -26,6 +26,19 @@ function SetupArrayRandomFunction(){
 			}
 }
 
+// Return a random element of the array if it's not equal to previuousValue
+function SetupArrayNonRepeatingRandomFunction(){
+	Array.prototype.notRepeatingRandom = function (previousValue) {
+		var result = this[Math.floor((Math.random()*this.length))];
+		if (result != previousValue) return result;
+			else {
+				result = this[Math.floor((Math.random()*this.length))];
+				if (result != previousValue) return result;
+			};
+		return undefined;
```

Comment:
Instead of return undefined please return a previous value (writing an error into console).